src/transformer_backends/gemini.rs

```diff
@@ -0,0 +1,239 @@
+        Self { config }
+    }
+
+    fn get_token(&self) -> anyhow::Result<String> {
+        if let Some(env_var_name) = &self.config.auth_token_env_var_name {
+            Ok(std::env::var(env_var_name)?)
+        } else if let Some(token) = &self.config.auth_token {
+            Ok(token.to_string())
+        } else {
+            anyhow::bail!(
+                "set `auth_token_env_var_name` or `auth_token` to use an Gemini compatible API"
+            )
+        }
+    }
+
+    async fn do_fim(&self, prompt: &FIMPrompt, params: GeminiRunParams) -> anyhow::Result<String> {
+        let client = reqwest::Client::new();
+        let token = self.get_token()?;
+        let res: serde_json::Value = client
+            .post(
+                self.config
+                    .completions_endpoint
+                    .as_ref()
+                    .context("must specify `gemini_endpoint` to use gemini")?
+                    .to_owned()
+                    + self.config.model.as_ref()
+                    + ":generateContent?key="
+                    + token.as_ref(),
+            )
+            .header("Content-Type", "application/json")
+            .json(&json!(
+                {
+                    "contents":[
+                        {
+                            "parts":[
+                                {
+                                    "text": prompt.prompt
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ))
+            .send()
+            .await?
+            .json()
+            .await?;
+        if let Some(error) = res.get("error") {
+            anyhow::bail!("{:?}", error.to_string())
+        } else if let Some(candidates) = res.get("candidates") {
+            Ok(candidates
+                .get(0)
+                .unwrap()
+                .get("content")
+                .unwrap()
+                .get("parts")
+                .unwrap()
+                .get(0)
+                .unwrap()
+                .get("text")
+                .unwrap()
+                .clone()
+                .to_string())
+        } else {
+            anyhow::bail!("Unknown error while making request to Gemini: {:?}", res);
+        }
+    }
+    async fn do_chat_completion(&self, prompt: &Prompt, params: Value) -> anyhow::Result<String> {
+        let client = reqwest::Client::new();
+        let token = self.get_token()?;
+        let res: serde_json::Value = client
+            .post(
+                self.config
+                    .chat_endpoint
+                    .as_ref()
+                    .context("must specify `gemini_endpoint` to use gemini")?
+                    .to_owned()
+                    + self.config.model.as_ref()
+                    + ":generateContent?key="
+                    + token.as_ref(),
+            )
+            .header("Content-Type", "application/json")
+            .json(&params)
```

Comment:
I'm not seeing where the value of `prompt` is being used here. Take a look at: https://github.com/SilasMarvin/lsp-ai/blob/cd46ecf61adc272491c25c4faf4b5f9d92c3d4a6/src/transformer_backends/ollama.rs#L142 In your case a FIM prompt type would just error, but we want to see if they have supplied messages for us. If they have supplied messages we would hit the [generateContent](https://ai.google.dev/api/rest/v1beta/models/generateContent) endpoint, otherwise we would hit the [generateMessage](https://ai.google.dev/api/rest/v1beta/models/generateMessage) endpoint.